datasets/udhr/udhr.py

```diff
@@ -0,0 +1,555 @@
+    "ron_2006": "Romanian (2006)",
+    "012": "(Bizisa)",
+    "roh_vallader": "Romansch (Vallader)",
+    "mly_arab": "Malay (Arabic)",
+    "swh": "Swahili",
+    "flm": "Chin, Falam",
+    "cha": "Chamorro",
+    "qvn": "Quechua, North Junín",
+    "arl": "Arabela",
+    "snk": "Soninke",
+    "quc": "K'iche', Central",
+    "dyo": "Jola",
+    "hlt": "Chin, Matu",
+    "ada": "Dangme",
+    "007": "Sãotomense",
+    "aka_akuapem": "Twi (Akuapem)",
+    "013": "(Mijisa)",
+    "yao": "Yao",
+    "lue": "Luvale",
+    "tgl_tglg": "Tagalog (Tagalog)",
+    "tso_ZW": "Tsonga (Zimbabwe)",
+    "uig_arab": "Uyghur (Arabic)",
+    "bod": "Tibetan, Central",
+    "bax": "Bamun",
+    "bba": "Baatonum",
+    "khk": "Mongolian, Halh (Cyrillic)",
+    "bam": "Bamanankan",
+    "kdh": "Tem",
+    "nno": "Norwegian, Nynorsk",
+    "ell_monotonic": "Greek (monotonic)",
+    "lug": "Ganda",
+    "ike": "Inuktitut, Eastern Canadian",
+    "nbl": "Ndebele",
+    "011": "Umbundu (011)",
+    "gjn": "Gonja",
+    "hni": "Hani",
+    "tpi": "Tok Pisin",
+    "roh_puter": "Romansch (Puter)",
+    "sme": "Saami, North",
+    "mfq": "Moba",
+    "arn": "Mapudungun",
+    "som": "Somali",
+    "slv": "Slovenian",
+    "qvm": "Quechua, Margos",
+    "cjk": "Chokwe",
+    "qwh": "Quechua, Huaylas Ancash",
+    "ces": "Czech",
+    "sco": "Scots",
+    "sag": "Sango",
+    "oss": "Osetin",
+    "huu": "Huitoto, Murui",
+    "zro": "Záparo",
+    "hms": "Hmong, Southern Qiandong",
+    "010": "Tetun Dili",
+    "nav": "Navajo",
+    "ppl": "Pipil",
+    "lat": "Latin",
+    "umb": "Umbundu",
+    "lld": "Ladin",
+    "kin": "Rwanda",
+}
+
+_WEBPAGE = "https://www.ohchr.org/EN/UDHR/Pages/UDHRIndex.aspx"
+
+# XML for meta-info about language and TXT for text
+_XML_DOWNLOAD_URL = "https://unicode.org/udhr/assemblies/udhr_xml.zip"
+_TXT_DOWNLOAD_URL = "https://unicode.org/udhr/assemblies/udhr_txt.zip"
+
+
+class UDHN(datasets.GeneratorBasedBuilder):
+    """Universal Declaration of Human Rights"""
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "text": datasets.Value("string"),
+                    "lang_key": datasets.Value("string"),
+                    "lang_name": datasets.Value("string"),
+                    "iso639-3": datasets.Value("string"),
+                    "iso15924": datasets.Value("string"),
+                }
+            ),
+            homepage=_WEBPAGE,
+        )
+
+    def _split_generators(self, dl_manager):
+        xml_dir = dl_manager.download_and_extract(_XML_DOWNLOAD_URL)
+        txt_dir = dl_manager.download_and_extract(_TXT_DOWNLOAD_URL)
+
+        return [
+            datasets.SplitGenerator(name=datasets.Split.TRAIN, gen_kwargs={"xml_dir": xml_dir, "txt_dir": txt_dir}),
+        ]
+
+    def _generate_examples(self, xml_dir, txt_dir):
+        for ilang, lang in enumerate(_LANGUAGES):
```

Comment:
There's a trick we can use here to reduce the size of the dummy data.
If we use `glob.glob` or `os.listdir` here instead of iterating through all the languages we can make this script compatible with dummy data that only contain 1 file.

note : when using glob or listdir make sure to always apply `sorted` to make sure the order is deterministic no matter the environment

Reply:
Oh ok that's a good idea. In that case I don't actually even need the massive `_LANGUAGES` list in the loading script at all. Do you think it's still worth having there for documentation purposes or just remove it?